pkg/api/controller.go

```diff
@@ -43,20 +43,37 @@ func (c *Controller) Run() error {
 	engine.Use(log.SessionLogger(c.Log), handlers.RecoveryHandler(handlers.RecoveryLogger(c.Log),
 		handlers.PrintRecoveryStack(false)))
 
-	c.ImageStore = storage.NewImageStore(c.Config.Storage.RootDirectory, c.Config.Storage.GC,
-		c.Config.Storage.Dedupe, c.Log)
-	if c.ImageStore == nil {
-		// we can't proceed without at least a image store
+	c.Router = engine
+	c.Router.UseEncodedPath()
+
+	// Assign global level image storage
+	c.ImageStore = make(map[string]*storage.ImageStore)
+
+	if c.Config.Storage.RootDirectory != "" {
+		c.ImageStore["/"] = storage.NewImageStore(c.Config.Storage.RootDirectory,
+			c.Config.Storage.GC, c.Config.Storage.Dedupe, c.Log)
+		// Enable extensions if extension config is provided
+		if c.Config != nil && c.Config.Extensions != nil {
+			ext.EnableExtensions(c.Config.Extensions, c.Log, c.Config.Storage.RootDirectory)
+		}
+	} else {
+		// we can't proceed without global storage
 		os.Exit(1)
```

Comment:
shouldnt we panic here with a proper log message?

Reply:
Yes, we should. I will add panic and log.